File: src/test/java/com/mikea/bayes/benchmarks/FactorProductBenchmark.java
package com.mikea.bayes.benchmarks;

import com.google.caliper.Runner;
import com.google.caliper.SimpleBenchmark;
import com.mikea.bayes.Factor;
import com.mikea.bayes.ProbabilitySpace;
import com.mikea.bayes.Var;
import com.mikea.bayes.VarSet;

import java.util.Collection;

import static com.google.common.collect.Lists.newArrayList;

/**
 * @author mike.aizatsky@gmail.com
 */
public class FactorProductBenchmark extends SimpleBenchmark {
    private Factor factor1;
    private Factor factor2;

    @Override
    protected void setUp() throws Exception {
        super.setUp();    //To change body of overridden methods use File | Settings | File Templates.

        ProbabilitySpace space = new ProbabilitySpace();
        Collection<Var> vars1 = newArrayList();
        Collection<Var> vars2 = newArrayList();

        for (int i = 0; i < 10; ++i) {
            vars1.add(space.newVar("a" + i, 2));
            vars2.add(space.newVar("b" + i, 2));
        }

        factor1 = Factor.constant(VarSet.newVarSet(vars1), 1);
        factor2 = Factor.constant(VarSet.newVarSet(vars2), 1);
    }

    @SuppressWarnings("UnusedDeclaration")
    public void time1(int reps) {
        for (int i = 0; i < reps; ++i) {
            Factor f = factor1.product(factor2);
        }
    }

    public static void main(String[] args) {
        Runner.main(FactorProductBenchmark.class, args);
    }
}

File: src/test/java/com/mikea/bayes/benchmarks/FactorProductBenchmark.scala
package com.mikea.bayes.benchmarks

import com.google.caliper.Runner
import com.google.caliper.SimpleBenchmark
import com.mikea.bayes.Factor

/**
 * @author mike.aizatsky@gmail.com
 */
object FactorProductBenchmark {
  def main(args: Array[String]) {
    Runner.main(classOf[FactorProductBenchmark], args)
  }
}

class FactorProductBenchmark extends SimpleBenchmark {
  protected override def setUp {
    ???
/*
    super.setUp
    val space: ProbabilitySpace = new ProbabilitySpace
    val vars1: Collection[Var] = newArrayList
    val vars2: Collection[Var] = newArrayList
    {
      var i: Int = 0
      while (i < 10) {
        {
          vars1.add(space.newVar("a" + i, 2))
          vars2.add(space.newVar("b" + i, 2))
        }
        ({
          i += 1; i
        })
      }
    }
    factor1 = Factor.constant(VarSet.newVarSet(vars1), 1)
    factor2 = Factor.constant(VarSet.newVarSet(vars2), 1)
*/
  }

  @SuppressWarnings(Array("UnusedDeclaration")) def time1(reps: Int) {
    {
      var i: Int = 0
      while (i < reps) {
        {
          val f: Factor = factor1.product(factor2)
        }
        ({
          i += 1; i
        })
      }
    }
  }

  private var factor1: Factor = null
  private var factor2: Factor = null
}